src/Microsoft.IdentityModel.Tokens/TokenValidationParameters.cs

```diff
@@ -353,6 +353,18 @@ public virtual ClaimsIdentity CreateClaimsIdentity(SecurityToken securityToken,
             return new ClaimsIdentity(authenticationType: AuthenticationType ?? DefaultAuthenticationType, nameType: nameClaimType ?? ClaimsIdentity.DefaultNameClaimType, roleType: roleClaimType ?? ClaimsIdentity.DefaultRoleClaimType);
         }
 
+        /// <summary>
+        /// This property represents the metadata (specifically the issuer and signing keys) retrieved by the <see cref="ConfigurationManager"/>.
+        /// It will be used along with the other properties set on the TokenValidationParameters in order to validate the token.
+        /// </summary>
+        public BaseConfiguration Configuration { get; set; }
```

Comment:
Should the Configuration always come from the ConfigurationManager (maybe the BaseConfigurationManager.CurrentConfiguration)? Right now they are two separate properties.

Reply:
I think setting this to always be equal to BaseConfigurationManager.CurrentConfiguration makes sense....that being said, this would mean there isn't any point in cloning the TVP to avoid potential modifications (as the value of this property will change once a configuration is retrieved).

Reply:
I added a comment which explains that this property could be set to either the LastKnownGoodConfiguration or the Current Configuration. https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1695/commits/f5b3504e5f237e293dce12f52e01e133ace89fe3